tests/utils/fixtures.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "faabric/util/memory.h"
+#include <faabric/redis/Redis.h>
+#include <faabric/scheduler/Scheduler.h>
+#include <faabric/util/testing.h>
+
+namespace tests {
+class BaseTestFixture
```

Comment:
This `BaseTestFixture` is here to capture the stuff I found we were repeating in all the existing features. We have to be careful not to make this a dumping ground, and only included the tidy-up that's actually going to get repeated elsewhere.

Reply:
I personally doubt that this is useful/desirable. The main point of fixtures was to remove the dumping ground `cleanFaabric()` had turned into. However, I feel we have just moved the burden of it to `BaseTestFixture`. 

For instance, why would the transport tests `redis.flushAll()`? I think there's value in knowing what you are modifying in your tests, and resetting just that.

Reply:
Yes, I did think that when creating this, the thing is with Redis and the scheduler for example, the snippets of clean-up would have to be sprinkled around so many places that it makes sense to avoid repetition. 

The solution I think should be a fixture per feature and have multiple inheritance in the tests, i.e. we have a `RedisFixture`, a `StateFixture`, a `MockClientsFixture` etc. I'll see if that works here...